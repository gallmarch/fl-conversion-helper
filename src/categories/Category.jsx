import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import CategoryToggle from './CategoryToggle';
import { setCategoryExpansion } from '../preferences/actions';

Category.propTypes = {
  preferences: PropTypes.shape({
    expansions: PropTypes.object.isRequired,
  }).isRequired,
  category: PropTypes.string.isRequired,
  categoryName: PropTypes.string.isRequired,
  children: PropTypes.node.isRequired,
};

function Category(props) {
  const { preferences: { expansions }, category } = props;
  const expanded = expansions[category];
  /* eslint-disable jsx-a11y/no-noninteractive-element-interactions */
  return (
    <div>
      <h3 onClick={() => toggleState(props)} className="flch-category-header">
        <CategoryToggle state={expanded} />
        {' '}
        <span className="flch-category-header__text">{props.categoryName}</span>
      </h3>
      {expanded && (
        <ul className="you_icon cf">
          {props.children}
        </ul>
      )}
    </div>
  );
  /* eslint-enable jsx-a11y/no-noninteractive-element-interactions */
}

function toggleState(props) {
  const { category, preferences: { expansions } } = props;
  const expanded = expansions[category];
  props.setCategoryExpansion({ category, expanded: !expanded });
}


function mapStateToProps(state) {
  return { preferences: state.preferences };
}

export default connect(mapStateToProps, { setCategoryExpansion })(Category);
