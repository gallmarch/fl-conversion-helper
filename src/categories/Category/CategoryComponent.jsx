import React from 'react';
import * as PropTypes from 'prop-types';
import CategoryToggle from '../CategoryToggle';

export default function CategoryComponent(props) {
  const { children, expanded, heading, onToggleExpanded } = props;
  if (!children.length) {
    return null;
  }
  return (
    <div>
      <div>
        <h3
          className="heading heading--2"
          style={{
            alignItems: 'center',
            display: 'flex',
            justifyContent: 'space-between',
          }}
        >
          {heading}
          <CategoryToggle expanded={expanded} onClick={onToggleExpanded} />
        </h3>
        {expanded && (
          <div className="row">
            <ul className="items items--inline inventory-group__items">
              {children}
            </ul>
          </div>
        )}
        <hr />
      </div>
    </div>
  );
}

CategoryComponent.propTypes = {
  children: PropTypes.node.isRequired,
  expanded: PropTypes.bool.isRequired,
  heading: PropTypes.string.isRequired,
  onToggleExpanded: PropTypes.func.isRequired,
};
