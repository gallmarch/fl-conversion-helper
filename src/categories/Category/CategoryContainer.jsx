import React, { Component } from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';

import { setCategoryExpansion } from '../../preferences/actions';
import CategoryComponent from './CategoryComponent';


export class CategoryContainer extends Component {
  constructor(props) {
    super(props);
    this.handleToggleExpanded = this.handleToggleExpanded.bind(this);
    this.state = { expanded: false };
  }

  handleToggleExpanded() {
    const { expanded, name } = this.props;
    this.props.setCategoryExpansion({
      category: name,
      expanded: !expanded,
    });
  }

  render() {
    const { children, ...rest } = this.props;
    return (
      <CategoryComponent
        {...rest}
        onToggleExpanded={this.handleToggleExpanded}
      >
        {children}
      </CategoryComponent>
    );
  }
}

CategoryContainer.propTypes = {
  children: PropTypes.node.isRequired,
  name: PropTypes.string.isRequired,
  expanded: PropTypes.bool.isRequired,
  heading: PropTypes.string.isRequired,
  setCategoryExpansion: PropTypes.func.isRequired,
};

const mapStateToProps = () => ({});

export default connect(mapStateToProps, { setCategoryExpansion })(CategoryContainer);
