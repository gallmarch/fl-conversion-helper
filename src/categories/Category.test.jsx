import React from 'react';
import renderer from 'react-test-renderer';

import { Category, mapStateToProps, toggleState } from './Category';

describe('Category', () => {
  it('matches its snapshot', () => {
    const category = 'category-id';
    const categoryName = 'category-name';
    const children = <li />;
    const preferences = { expansions: { [category]: true } };
    const component = renderer.create(<Category
      category={category}
      categoryName={categoryName}
      preferences={preferences}
    >
      {children}
    </Category>);

    expect(component).toMatchSnapshot();
  });

  describe('#toggleState', () => {
    const category = 'category-id';
    it('flips true to false', () => {
      const initialValue = true;
      const props = {
        category,
        preferences: { expansions: { [category]: initialValue } },
        setCategoryExpansion: jest.fn(),
      };
      toggleState(props);
      expect(props.setCategoryExpansion).toHaveBeenCalledWith({ category, expanded: false });
    });

    it('flips true to false', () => {
      const initialValue = false;
      const props = {
        category,
        preferences: { expansions: { [category]: initialValue } },
        setCategoryExpansion: jest.fn(),
      };
      toggleState(props);
      expect(props.setCategoryExpansion).toHaveBeenCalledWith({ category, expanded: true });
    });
  });

  describe('#mapStateToProps', () => {
    it('returns the preferences property', () => {
      const preferences = { key: 'expected-value' };
      expect(mapStateToProps({ preferences })).toEqual({ preferences });
    });
  });
});
