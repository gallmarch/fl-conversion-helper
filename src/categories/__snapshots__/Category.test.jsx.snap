// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Category matches its snapshot 1`] = `
<div>
  <h3
    className="flch-category-header"
    onClick={[Function]}
  >
    <span
      className="contract"
    >
      -
    </span>
     
    <span
      className="flch-category-header__text"
    >
      category-name
    </span>
  </h3>
  <ul
    className="you_icon cf"
  >
    <li />
  </ul>
</div>
`;
